### PR TITLE
[CARBONDATA-1214]changing the delete syntax as in the hive.

### PR DESCRIPTION
**Pull request by ravikiran23**
Problem  : The syntax of the carbon delete by id and date is not compatible to hive.

Solution  : Making the delete syntax compatible to hive.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2659/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/88/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/576/<h2>Failed Tests: <span class='status-failure'>2</span></h2><h3><a name='carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-core' /><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/576/org.apache.carbondata$carbondata-core/testReport'>carbondata-pr-spark-1.6/org.apache.carbondata:carbondata-core</a>: <span class='status-failure'>2</span></h3><ul><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/576/org.apache.carbondata$carbondata-core/testReport/org.apache.carbondata.core.writer/CarbonFooterWriterTest/testWriteFactMetadata/'><strong>org.apache.carbondata.core.writer.CarbonFooterWriterTest.testWriteFactMetadata</strong></a></li><li><a href='https://builds.apache.org/job/carbondata-pr-spark-1.6/576/org.apache.carbondata$carbondata-core/testReport/org.apache.carbondata.core.writer/CarbonFooterWriterTest/testReadFactMetadata/'><strong>org.apache.carbondata.core.writer.CarbonFooterWriterTest.testReadFactMetadata</strong></a></li></ul>

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/carbondata-pr-spark-1.6/577/

File: examples/spark/src/main/scala/org/apache/carbondata/examples/DataManagementExample.scala [hunk 1]
```diff
@@ -45,7 +45,7 @@ object DataManagementExample {
     cc.sql("SHOW SEGMENTS FOR TABLE t3 ").show
 
     // delete the first segment
-    cc.sql("DELETE SEGMENT 0 FROM TABLE t3")
+    cc.sql("delete from table t3 where segment.id in (0)")
```
Comment (chenliang613) [inline examples/spark/src/main/scala/org/apache/carbondata/examples/DataManagementExample.scala]: please use uppercase for "DELETE,FROM ,TABLE, WHERE, IN "

**Comment by chenliang613** [comment]
@ravikiran23  please update all documents accordingly.

**Comment by gvramana** [comment]
@ravikiran23 Please update the document that old syntax of "DELETE SEGMENT" is deprecated from 1.2 version and new syntax is added.
@chenliang613 please comment on the same.

**Comment by ManoharVanam** [comment]
@chenliang613 documents handling will be done by @sgururajshetty in separate jira.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder/2932/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder/346/

**Comment by gvramana** [comment]
LGTM

**Comment by sgururajshetty** [comment]
Document updated with PR 1141

**Pull request closed without merging**